src/plot_api/plot_api.js

```diff
@@ -205,7 +205,7 @@ Plotly.plot = function(gd, data, layout, config) {
             if(trace.visible !== true || !trace._module.colorbar) {
                 Plots.autoMargin(gd, 'cb' + trace.uid);
             }
-            else trace._module.colorbar(gd, cd);
+            else if(trace.marker.colorbar) trace._module.colorbar(gd, cd);
```

Comment:
I made the changes and run test locally. It still fails in the contour/heatmap traces assertion on colorbar, https://github.com/plotly/plotly.js/blob/master/test/jasmine/tests/plot_interact_test.js#L223 and https://github.com/plotly/plotly.js/blob/master/test/jasmine/tests/plot_interact_test.js#L283. Both ``countColorBars`` function returns 0. 

I will take a look. And it would be great if you could shed some lights here.